File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCount=0;
        int currCount=0;

        for(int num:nums){
            if(num==1){
                currCount++;
                maxCount=max(maxCount, currCount);
            }
            if(num==0){
                currCount=0;
            }
        }
        return maxCount;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let maxVal=0;
    let currVal=0;

    for(let num of nums){
        if(num===1){
            currVal++;
            maxVal=Math.max(maxVal, currVal);
        }
        if(num==0){
            currVal=0;
        }
    }
    return maxVal;
};